tests/dummy/app/controllers/update-for-reposition.js

```diff
@@ -1,43 +1,44 @@
 import Controller from '@ember/controller';
-import { set } from '@ember/object';
+import { action, set } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
 
-export default Controller.extend({
-  rows: null,
-  isLoading: true,
+export default class UpdateForRepositionController extends Controller {
+  @tracked
+  rows = [];
 
-  init() {
-    this._super(...arguments);
-    set(this, 'rows', []);
-  },
+  @tracked
+  isLoading = true;
 
-  actions: {
-    fetchData() {
-      setTimeout(() => {
-        if (this.isDestroying) {
-          return;
-        }
+  @action
+  fetchData() {
+    setTimeout(() => {
+      if (this.isDestroying) {
+        return;
+      }
 
-        set(this, 'rows', [
-          {
-            text: 'random text',
-            number: 1,
-          },
-          {
-            text: 'random text',
-            number: 2,
-          },
-          {
-            text: 'random text',
-            number: 3,
-          },
-        ]);
-        set(this, 'isLoading', false);
-      }, 1000);
-    },
+      this.rows = [
+        {
+          text: 'random text',
+          number: 1,
+        },
+        {
+          text: 'random text',
+          number: 2,
+        },
+        {
+          text: 'random text',
+          number: 3,
+        },
+      ];
+      this.isLoading = false;
+    }, 1000);
+  }
 
-    deleteData() {
-      set(this, 'isLoading', true);
-      set(this, 'rows', []);
-    },
-  },
-});
+  @action
+  deleteData() {
+    if (!this.isLoading) {
```

Comment:
Slight modification as in tracked world without this guard we'd end up with infinite loop for updateFor attribute.